rmw_dds_common/test/test_graph_cache.cpp

```diff
@@ -1550,3 +1551,64 @@ TEST(test_graph_cache, bad_arguments)
     rcutils_reset_error();
   }
 }
+
+TEST(test_graph_cache, get_writers_info_by_topic_maybe_fail)
+{
+  RCUTILS_FAULT_INJECTION_TEST(
+  {
+    GraphCache graph_cache;
+    add_participants(graph_cache, {"participant1"});
+    add_entities(
+      graph_cache,
+    {
+      // topic1
+      {"reader1", "participant1", "topic1", "Str", true},
+      {"writer1", "participant1", "topic1", "Str", false},
+    });
+    add_nodes(
+      graph_cache, {
+      {"participant1", "ns1", "node1"},
+      {"participant1", "ns1", "node2"},
+      {"participant1", "ns2", "node1"}});
+
+    rmw_topic_endpoint_info_array_t info = rmw_get_zero_initialized_topic_endpoint_info_array();
+
+    rcutils_allocator_t allocator = rcutils_get_default_allocator();
+    rmw_ret_t ret = graph_cache.get_writers_info_by_topic(
+      "topic1",
+      identity_demangle,
+      &allocator,
+      &info);
+    if (RMW_RET_OK == ret) {
+      ret = rmw_topic_endpoint_info_array_fini(&info, &allocator);
```

Comment:
@brawner should this be:
```suggestion
      EXPECT_EQ(RWM_RET_OK, rmw_topic_endpoint_info_array_fini(&info, &allocator));
```
instead? Also, what if fault injection occurs here?

Reply:
Actually, it will fail during a fault injection test because there will be at least one hook in there. I added a check on the return of fini just in case it needs be done a second time.